tinygrad/codegen/kernel.py

```diff
@@ -352,7 +352,8 @@ def buf_index(src: LazyOp) -> Optional[int]:
           # TODO: apply tc even if the sources are not from LOAD
           if src.op is BufferOps.LOAD and src.arg.dtype == tc.dtype_in: return self.bufs.index(cast(MemBuffer, src.arg))
           try:
-            if opt_level >= 1 and src.op is UnaryOps.CAST and src.arg[0] == tc.dtype_in: return self.bufs.index(cast(MemBuffer, src.src[0].arg))
+            if opt_level >= 1 and src.op in [UnaryOps.CAST, UnaryOps.BITCAST] and src.arg == tc.dtype_in:
```

Comment:
this is tc too...

Reply:
In the old version we didn’t filter based on cast arg true/false, so didn’t CAST also mean bitcast?

Reply:
Just fixed it.